pkg/controller.v2/controller_status.go

```diff
@@ -431,6 +193,11 @@ func isFailed(status tfv1alpha2.TFJobStatus) bool {
 // If the condition that we are about to add already exists
 // and has the same status and reason then we are not going to update.
 func setCondition(status *tfv1alpha2.TFJobStatus, condition tfv1alpha2.TFJobCondition) {
+	//Do nothing if condition Failed
+	if isFailed(*status) {
```

Comment:
Besides this, I think we should set the condition Running to false when the job is succeeded or failed.